experimental/bigint.md

```diff
@@ -0,0 +1,12 @@
+# [BigInt](https://github.com/tc39/proposal-bigint)
+
+```js
+extend interface Literal {
+  value: string | boolean | null | number | RegExp | bigint;
+}
+
+interface BigIntLiteral <: Literal {
+  value: bigint;
+  bigint: string;
```

Comment:
How about `raw` and `cooked`? (i.e.`raw` would be `"12345n"` and cooked would be `12345n` (as a `BigInt`)?

Reply:
`cooked` only exists with template literals. `raw` is not part of the spec for (non-template) literals (see #14).

As for platforms without BigInt support, I've been asking this myself, too. Babylon doesn't emit the value, so I'm not aware of any precedents. I can think of a) emitting null b) casting to a regular number, which is lossy.

Reply:
FYI “platforms without `BigInt` support” includes (and will include, for all eternity<sup>[*citation needed*]</sup>) JSON.

Reply:
You are right, just as regular expressions. There's a discussion in #64 about JSON.

Reply:
I think emitting `null` if bigint is not in safe integer space makes sense and would be in line with regular expression literals.

Reply:
Why safe integer space? Use a bigint if the platform supports it (regardless of magnitude), `null` otherwise.

Reply:
I've made `value: bigint | null` for now, but I would still prefer to drop or phase out `value` as explained in https://github.com/estree/estree/issues/64#issuecomment-357434955.